generators/spring-boot/templates/build.gradle.jhi.spring-boot.ejs

```diff
@@ -0,0 +1,29 @@
+<%#
+ Copyright 2013-2021 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+<&_ if (fragment.pluginsSection) { -&>
+  id 'org.springframework.boot' version '<%= SPRING_BOOT_VERSION %>'
+  id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+<&_ } -&>
+<&_ if (fragment.dependenciesSection) { -&>
+  implementation 'org.springframework.boot:spring-boot-starter'
```

Comment:
`generator/buildTool/buildToolDependencyFactory.js#toVendorDependency('org.springframework.boot', 'spring-boot-starter', 'compile')` -> `generator/buildTool/gradle/gradleBuildToolDependencyFactory.js#toVendorDependency(`?

Not sure if it make sense or will simplify anything, but with that, most of the fragment can be common to both vendors.

Reply:
For now I am avoiding implementing apis in the generator for now, but we may implement some unified apis to replace current needle apis.

https://github.com/jhipster/generator-jhipster/blob/fc5d582de6bfcef89dbae42e632a5299a48687a7/generators/generator-base.js#L958-L1170